src/components/TransactionResult.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  AlertTitle,
+  Flex,
+} from "@chakra-ui/react";
+
+export enum ActionName {
+  Deposit = "Deposit",
+  Withdraw = "Withdraw",
+  Approve = "Approve",
+  Associate = "Associate",
+  Mint = "Mint",
+  Claim = "Claim",
+}
+
+const getAlertTitle = (isSuccess = true, actionType?: ActionName) => {
+  return `${actionType} ${isSuccess ? "success" : "error"}`;
+};
+
+export const TransactionResult = ({
+  actionName,
+  transactionResult,
+  transactionError,
+}: {
+  actionName: ActionName | undefined;
+  transactionResult: `0x${string}` | undefined;
+  transactionError: Error | null | undefined;
+}) => {
+  if (!actionName || !(transactionResult && transactionError)) {
```

Comment:
This check confuses me tbh.
When we dont have a name?
Should we also have 2 cases, when we have result but no error, and have no result but have en error?

Reply:
> This check confuses me tbh. When we dont have a name? Should we also have 2 cases, when we have result but no error, and have no result but have en error?

You right here, we always should have a name.